### DISCUSSION TITLE
[Feature] Drop http keep-alive support in machine learning container

### DISCUSSION THREAD

**Discussion by rkojedzinszky** [immich-app/immich#12064] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I have deployed immich on a k8s cluster. I've set up horizontal pod autoscaler for machine-learning component. Howewer, as that container serves requests over http/1.1, the connections remain open, and basically, the newly created machine-learning pods dont receive any new requests, only the old one does.

Thus, I would suggest to disable http keep-alive in the ML server. I've tested this in my case, it is working, and as expected, When k8s autoscaler starts new PODs, they start receiving new requests, effectively speeding up the processing.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by rkojedzinszky** [comment]
By not using HTTP keep-alives, all TCP connections get closed after a request, and this all new requests will go through load-balancing, if one exists. As ML processing may take time, the little overhead of establishing a new TCP connection upon each request is negligible.

**Reply by troyliu0105** [reply]
May some 3rd-party load balancer can solve your problem, like traefik's loadBalancer?

**Reply by rkojedzinszky** [reply]
I would not like to place another intermediary component between microservice and ml components. Howewer, I am already happy with the approved [PR](https://github.com/immich-app/immich/pull/12065).
